executor/include/executor/trait/is_instance_of_base.hpp

```diff
@@ -14,18 +14,21 @@
 
 namespace executor {
 
-template <template <typename...> class Type, typename Executor>
-struct is_instance_of_base : std::false_type {};
+template <typename Executor, template <typename...> class Type>
+struct is_instance_of_base_impl : std::false_type {};
 
 template <template <typename...> class Type, typename... Args>
-struct is_instance_of_base<Type, Type<Args...>> : std::true_type {};
+struct is_instance_of_base_impl<Type<Args...>, Type> : std::true_type {};
 
-template <template <typename...> class Type, typename Executor>
+template <typename Executor, template <typename...> class... Type>
+using is_instance_of_base = executor::disjunction<is_instance_of_base_impl<Executor, Type>...>;
```

Comment:
The Executor coming before Type... might require a small change in name of `is_instance_of_base` into `is_instance_of_all` or `is_instance_of_any` (depending on conjunction/disjunction)

Reply:
True. I think we should also finalize the name for `is_instance_of_base` to whatever is more descriptive.
@SergioRAgostinho

Reply:
Please take a decision @shrijitsingh99, let him comment on the decision :smile:

Reply:
Renamed to `InstanceOfBase`